pkgs/build-support/dotnet/add-nuget-deps/default.nix

```diff
@@ -20,16 +47,17 @@ let
   attrs = if builtins.isFunction fnOrAttrs then fnOrAttrs finalAttrs else fnOrAttrs;
 
   deps =
-    if (nugetDeps != null) then
-      if lib.isDerivation nugetDeps then
-        [ nugetDeps ]
-      else if lib.isList nugetDeps then
-        nugetDeps
-      else
-        assert (lib.isPath nugetDeps);
-        callPackage nugetDeps { fetchNuGet = fetchNupkg; }
+    if nugetDeps == null then
+      [ ]
+    else if lib.isDerivation nugetDeps then
+      [ nugetDeps ]
+    else if lib.isList nugetDeps then
+      nugetDeps
+    else if lib.hasSuffix ".json" nugetDeps then
```

Comment:
Theoretically, someone may use a directory with a `default.nix` as their nugetDeps... I guess that's a bit contrived/unlikely, but this implementation supports it while checking for `.nix` does not.

Either way, it should be consistent. You're right.

Reply:
Yeah, I'm not bothered about which one is the default:

> Theoretically, someone may use a directory with a default.nix as their nugetDeps

I don't think I ever tested this.  I assume it would break `fetch-deps`, even if it could still be used...